v2/AdminCore.Services/EventService.cs

```diff
@@ -0,0 +1,404 @@
+      var events = _databaseContext.EventRepository.Get(x => x.EventStatus.EventStatusId == (int)eventStatus
+                                                             && x.EventType.EventTypeId == (int)eventType);
+      if (!events.Any())
+      {
+        responseMessage.Payload = _mapper.Map<IList<EventDto>>(events);
+        responseMessage.Status = MessageConstants.MsgStatusSuccess;
+      }
+      else
+      {
+        responseMessage.Status = MessageConstants.MsgStatusNoRecords;
+      }
+      return responseMessage;
+    }
+
+    public ResponseMessage<IList<EventDto>> GetByType(EventTypes eventType)
+    {
+      ResponseMessage<IList<EventDto>> responseMessage = new ResponseMessage<IList<EventDto>>(null);
+      var events = _databaseContext.EventRepository.Get(x => x.EventType.EventTypeId == (int)eventType);
+      if (!events.Any())
+      {
+        responseMessage.Payload = _mapper.Map<IList<EventDto>>(events);
+        responseMessage.Status = MessageConstants.MsgStatusSuccess;
+      }
+      else
+      {
+        responseMessage.Status = MessageConstants.MsgStatusNoRecords;
+      }
+      return responseMessage;
+    }
+
+    public ResponseMessage<EventDto> CreateEvent(EventDto eventDto)
+    {
+      ICollection<EventDateDto> eventDates = new List<EventDateDto>();
+      eventDates = SplitEventIfFallsOnAWeekend(eventDto, eventDto.EventDates.Last().EndDate, eventDates);
+      eventDto.EventDates = eventDates;
+
+      ResponseMessage<EventDto> responseMessage = new ResponseMessage<EventDto>(null);
+      var existingEvent = GetEventById(eventDto.EventId);
+
+      if (existingEvent == null)
+      {
+        responseMessage.Payload = eventDto;
+        responseMessage.Status = MessageConstants.MsgStatusSuccess;
+        Save(_mapper.Map<Event>(eventDto));
+        return responseMessage;
+      }
+
+      responseMessage.Status = MessageConstants.MsgStatusFailed;
+      return responseMessage;
+    }
+
+    public ICollection<EventDateDto> SplitEventIfFallsOnAWeekend(EventDto eventDto, DateTime originalEndDate,
+      ICollection<EventDateDto> eventDates)
+    {
+      DateTime startDate = eventDto.EventDates.First().StartDate;
+      DateTime endDate = eventDto.EventDates.Last().EndDate;
+      foreach (DateTime day in EachDay(startDate, endDate))
+      {
+        if (day.DayOfWeek == DayOfWeek.Saturday)
+        {
+          // Set end date
+          eventDates.Add(SetEndDateForNewEvent(eventDto, eventDates, day));
+          // Create new event
+          EventDto nextEvent = MapEventDto(eventDto, originalEndDate, day, 2);
+          // Check again
+          SplitEventIfFallsOnAWeekend(nextEvent, nextEvent.EventDates.Last().EndDate, eventDates);
+          break;
+        }
+      }
+
+      if (eventDto.EventDates.Last().EndDate == originalEndDate)
+      {
+        EventDateDto eventDateDto = new EventDateDto
+        {
+          StartDate = eventDto.EventDates.First().StartDate,
```

Comment:
First isn't safe to use. If you want to use first you should also use null propaganda. eventDto.EventDates.First?.StartDate

Reply:
I would even use FirstOrDefault as it won't throw an InvalidOperationException.

eventDto.EventDates.FirstOrDefault?.StartDate